components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/Utils.kt

```diff
@@ -70,14 +73,53 @@ suspend fun handleFxaExceptions(logger: Logger, operation: String, block: () ->
     })
 }
 
-private fun shouldThrow(e: FxaException): Boolean {
-    return when (e) {
-        // Throw on panics
-        is FxaPanicException -> true
-        // Don't throw for recoverable errors.
-        is FxaNetworkException, is FxaUnauthorizedException, is FxaUnspecifiedException -> false
-        // Throw on newly encountered exceptions.
-        // If they're actually recoverable and you see them in crash reports, update this check.
-        else -> true
+/**
+ * Simplified version of Kotlin's inline class version that can be used as a return value.
+ */
+internal sealed class Result<out T> {
+    data class Success<out T>(val value: T) : Result<T>()
+    object Failure : Result<Nothing>()
+}
+
+internal suspend fun <T> withRetries(logger: Logger, retryCount: Int, block: suspend () -> T): Result<T> {
+    var attempt = 0
+    var res: T? = null
+    while (attempt < retryCount && (res == null || res == false)) {
+        attempt += 1
+        logger.info("attempt $attempt/$retryCount")
```

Comment:
nit: Now that this is in Utils, should we prefix the log statement so it's clear that it's from `withRetries`?